### PR TITLE
Resolve fixme in test qp_olap_windowerr, temp_relation

### PR DESCRIPTION
**Pull request by QingMa-gp**
This PR contains the following changes:
1.  Resolve fixme in test qp_olap_windowerr
    > GPDB_12_MERGE_FIXME: delete the _optimizer answer file
    > The only diff between the planner and ORCA answer files are:
    > INFO:  GPORCA failed to produce a plan, falling back to planner
    > DETAIL:  Feature not supported: window frame RANGE with OFFSET PRECEDING
    
    This commit deleted `qp_olap_windowerr_optimizer.out`.
2. Resolve fixme in test temp_relation
      > GPDB_12_MERGE_FIXME: Validate after we drop the same table, no
      > files are left behind. Current function
      > temp_relfilenode_primaries() relies on pg_class to find the
      > relfilenode and hence is not suitable to perform this validation.
      
      By setting guc `gp_select_invisible` and `autovacuum`, we can get
      the relfilenode of the table that has been dropped.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/regress/output/temp_relation.source [hunk 1]
```diff
@@ -9,7 +9,7 @@ STRICT STABLE LANGUAGE SQL AS
     (SELECT 't_' || relfilenode AS relfilenodename FROM pg_class WHERE relname=rel),
   db_relfilenodenames AS
     (SELECT pg_ls_dir('./base/' || t.dboid || '/') AS relfilenodename
-      FROM (SELECT oid AS dboid FROM pg_database WHERE datname = 'regression') t)
+      FROM (SELECT oid AS dboid FROM pg_database WHERE datname = 'regression' and xmax = 0) t)
```
Comment (Aegeaner) [inline src/test/regress/output/temp_relation.source:12]: Can't we just remember the `relfilenodename` of the table to be dropped before the validation? The current approach seems too tricky for this task.

Comment (QingMa-gp) [reply src/test/regress/output/temp_relation.source:12]: `relfilenodename` of temporary table may not be equal on Coordinator and segments. I didn't think of a proper way to store this value.

File: src/test/regress/output/temp_relation.source [hunk 2]
```diff
@@ -37,6 +37,18 @@ STRICT STABLE LANGUAGE SQL AS
   SELECT temp_relfilenode_segments(rel) AS segno
   ORDER BY segno;
   $fn$;
+-- The following gucs are set to get refilenode from pg_class after we droped temp_table.
+alter system set autovacuum = off;
+select gp_segment_id, pg_reload_conf() from gp_id union select gp_segment_id, pg_reload_conf() from gp_dist_random('gp_id');
```
Comment (haolinw) [inline src/test/regress/output/temp_relation.source:42]: Is that necessary to add ORDER BY gp_segment_id to make the result order solid ?

Comment (kainwen) [reply src/test/regress/output/temp_relation.source:42]: > Is that necessary to add ORDER BY gp_segment_id to make the result order solid ?

Current Test framework  will ignore order.

Comment (QingMa-gp) [reply src/test/regress/output/temp_relation.source:42]: Regress will sort the query output for all select statements that do not have an `ORDER BY`.

Comment (haolinw) [reply src/test/regress/output/temp_relation.source:42]: Got it, thank you both. The code looks good to me.

**Review by haolinw — APPROVED** [review]

**Comment by interma** [comment]
PR pipeline is failed, looks it's related to this PR?

**Comment by QingMa-gp** [comment]
> PR pipeline is failed, looks it's related to this PR?

@interma. Yes. I am trying to resolve it.

**Pull request merged**